src/verifast1.ml

```diff
@@ -3551,7 +3999,18 @@ module VerifyProgram1(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
         | ((Int (_, _)|Float|Double|LongDouble), (Float|Double|LongDouble)) -> floating_point_fun_call_expr funcmap (expr_loc w) t0 ("of_" ^ identifier_string_of_type t) [TypedExpr (w, t)]
         | ((Float|Double|LongDouble), (Int (_, _))) -> floating_point_fun_call_expr funcmap (expr_loc w) t0 ("of_" ^ identifier_string_of_type t) [TypedExpr (w, t)]
         | (ObjType ("java.lang.Object", []), ArrayType _) when isCast -> w
-        | _ ->
+        | _ when inAnnotation <> Some(true) && language = Java ->
+          let inferredTypes = 
+            if proper_type t then
+              []
+            else begin try infer_type (expr_loc e) [t] [t0] 

```

Comment:
Fout. Je gaat er hier van uit dat een RealTypeParam een nog te infereren type is. Maar een RealTypeParam kan ook gewoon een echte RealTypeParam zijn!!! Bv.: in `static <T, U> U foo(T x) { return x; }` is het niet OK om T gewoon te vervangen door U om het typechecken van `return x;` te doen slagen!